### PR TITLE
add benchmarks

### PR DESCRIPTION
**Pull request by yogsototh**
Added a benchmark system.

To launch it

~~~
lein perforate
~~~

It will start a CTIA server and benchmark actor creation between small and big actor entity.
It takes a _very_ long time (about 2 min) to finish. But in the end you'll have a resume like this:

~~~
Goal:  Create Actor
-----
Case:  :small-actor
Evaluation count : 23580 in 60 samples of 393 calls.
             Execution time mean : 2.545941 ms
    Execution time std-deviation : 86.434541 µs
   Execution time lower quantile : 2.482845 ms ( 2.5%)
   Execution time upper quantile : 2.787981 ms (97.5%)

Found 2 outliers in 60 samples (3.3333 %)
	low-severe	 2 (3.3333 %)
 Variance from outliers : 20.6060 % Variance is moderately inflated by outliers

Case:  :big-actor
Evaluation count : 5280 in 60 samples of 88 calls.
             Execution time mean : 11.069992 ms
    Execution time std-deviation : 519.617484 µs
   Execution time lower quantile : 10.623136 ms ( 2.5%)
   Execution time upper quantile : 12.150540 ms (97.5%)

Found 2 outliers in 60 samples (3.3333 %)
	low-severe	 1 (1.6667 %)
	low-mild	 1 (1.6667 %)
 Variance from outliers : 33.5521 % Variance is moderately inflated by outliers
~~~

I'm not familiar with this library, apparently we can also use fixtures. But I believe that using the `setup` and `cleanup` helps not counting the start and shutdown of the server in the tests time.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -104,7 +105,8 @@
                                      "test/resources/hooks/hook-example-0.1.0-SNAPSHOT.jar"]}
              :prepush {:plugins [[yogsototh/lein-kibit "0.1.6-SNAPSHOT"]
                                  [lein-bikeshed "0.3.0"]]}}
-  :plugins [[lein-shell "0.5.0"]]
+  :plugins [[lein-shell "0.5.0"]
+            [perforate "0.3.4"]]
```
Comment (quoll) [inline project.clj]: Do we want to consider setting up:
`:perforate { :environments [{:name :default
                                             :namespaces [ctia.http.routes.actor-bench]`
This sets up a template to make expansion of the benchmarking environment easier in the future (if there is the possibility of expanding it).

**Comment by polygloton** [comment]
Disable the metric services in the fixture-properties:clean test helper. :-)

File: project.clj (hunk 1 above)
Comment (yogsototh) [inline project.clj]: For now, I only created one benchmark test, but I imagine that yes, we certainly want to focus only on certain benchmarks. So I'll add this.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,10 @@ production, unless you ensure access to the NREPL port is restricted.
 
 This will start up with non-persistent in-memory storage only.
 
+**Remark**:
+if you use java 1.7, you might need to add `-XX:MaxPermSize=256m`
+to java options.
```
Comment (polygloton) [inline README.md]: We have this in :jvm-opts in project.clj.  Is it not the same?

Comment (yogsototh) [inline README.md]: Yes while doing this PR I was annoyed by the double warning each time I launch a lein run. So I moved the `-XX:MaxPermSize=256` in the travis-ci as an environment variable. So the warning is only visible during travis CI tests with java-1.8. (this is the first commit of this PR)

**Comment by gbuisson** [comment]
those are the accurate bench results after my fix, previously the ES http setup was mistakenly used on all cases:

```
Goal:  Create Actor
-----
Case:  :big-actor-atom-store
Evaluation count : 5760 in 60 samples of 96 calls.
             Execution time mean : 10.398050 ms
    Execution time std-deviation : 197.636183 µs
   Execution time lower quantile : 10.058094 ms ( 2.5%)
   Execution time upper quantile : 10.815132 ms (97.5%)

Found 2 outliers in 60 samples (3.3333 %)
	low-severe	 2 (3.3333 %)
 Variance from outliers : 7.8327 % Variance is slightly inflated by outliers

Case:  :big-actor-es-store
Evaluation count : 4080 in 60 samples of 68 calls.
             Execution time mean : 14.767150 ms
    Execution time std-deviation : 281.124055 µs
   Execution time lower quantile : 14.349086 ms ( 2.5%)
   Execution time upper quantile : 15.402148 ms (97.5%)

Found 2 outliers in 60 samples (3.3333 %)
	low-severe	 1 (1.6667 %)
	low-mild	 1 (1.6667 %)
 Variance from outliers : 7.8343 % Variance is slightly inflated by outliers

Case:  :big-actor-es-store-native
Evaluation count : 4440 in 60 samples of 74 calls.
             Execution time mean : 13.731146 ms
    Execution time std-deviation : 327.521919 µs
   Execution time lower quantile : 13.353692 ms ( 2.5%)
   Execution time upper quantile : 14.239146 ms (97.5%)

Found 1 outliers in 60 samples (1.6667 %)
	low-severe	 1 (1.6667 %)
 Variance from outliers : 11.0437 % Variance is moderately inflated by outliers
```

**Comment by gbuisson** [comment]
LGTM

**Pull request merged**